### PR TITLE
Refactoring using OrderAmountUpdater

### LINKED ISSUE #18432: Refactor DeleteProductFromOrderHandler
- [x] #18564 : Refactor two handlers : 
- src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php
- src/Adapter/Order/CommandHandler/DeleteProductFromOrderHandler.php

because updateOrder are nearly the same.

- [ ] Refactor handler src/Adapter/Order/CommandHandler/DeleteProductFromOrderHandler.php for using src/Adapter/Order/Refund/OrderProductRemover.php as service.

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Refactoring using OrderAmountUpdater
| Type?         | refacto
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #18432 
| How to test?  | Tests must be :green_circle: 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18564)

### CODE REVIEW

File: src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php [hunk 1]
```diff
@@ -82,15 +78,29 @@ final class AddProductToOrderHandler extends AbstractOrderHandler implements Add
      */
     private $contextStateManager;
 
+    /**
+     * @var ContextStateManager
```
Comment (sowbiba) [inline src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php]: wrong type

File: src/Adapter/Order/OrderAmountUpdater.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+class OrderAmountUpdater
+{
+    /**
+     * @param Cart $cart
+     * @param Currency $currency
+     * @param Order $order
+     * @param OrderDetail $orderDetail
+     */
+    public function updateAmount(
```
Comment (sowbiba) [inline src/Adapter/Order/OrderAmountUpdater.php]: the method name can only be update(...) as the class name is explicit

File: src/Adapter/Order/OrderAmountUpdater.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
```
Comment (sowbiba) [inline src/Adapter/Order/OrderAmountUpdater.php:26]: declare strict-types

File: src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php (hunk 1 above)
Comment (Progi1984) [reply src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php]: Fixed

File: src/Adapter/Order/OrderAmountUpdater.php (hunk 2 above)
Comment (Progi1984) [reply src/Adapter/Order/OrderAmountUpdater.php]: Fixed

File: src/Adapter/Order/OrderAmountUpdater.php (hunk 3 above)
Comment (Progi1984) [reply src/Adapter/Order/OrderAmountUpdater.php:26]: Fixed

**Review by matthieu-rolland — APPROVED** [review]

**Review by sowbiba — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @Progi1984

**Pull request merged**